engine/class_modules/paladin/sc_paladin.hpp

```diff
@@ -1104,6 +1112,15 @@ struct holy_power_consumer_t : public Base
 
     ab::execute();
 
+    if (
+        hp_used > 0 &&
+        p -> legendary.from_dusk_till_dawn -> ok() &&
+        p -> resources.current[ RESOURCE_HOLY_POWER ] == p -> legendary.from_dusk_till_dawn -> effectN( 2 ).base_value()
+      )
+    {
+      p -> buffs.blessing_of_dusk -> trigger();
+    }
+
```

Comment:
Similarly to blessing of dawn, this could be move to paladin_t::resource_loss() after player_t::resource_loss() and would only need to check whether the player is at 0 and if the player actually lost any resource this action.